src/tasks/job_board_interface.h

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "tasks/task.h"
+
+#include <mutex>
+#include <queue>
+#include <thread>
+
+namespace ccf::tasks
+{
+  struct IJobBoard
+  {
+    virtual void add_task(Task&& t) = 0;
+    virtual Task get_task() = 0;
```

Comment:
I think get_task() could do with some doc, is this getting the next executable task? Any next executable task?